jobserver/api/releases.py

```diff
@@ -43,44 +44,139 @@ def perform_create(self, serializer):
             logger.exception("Failed to notify slack")
 
 
-class ReleaseUploadAPI(APIView):
+def validate_upload_access(request, workspace):
+    """Validate the request can upload releases for this workspace.
+
+    This validation uses backend authentication to authenticate the user, but
+    then checks that user has the correct permissons."""
+    # authenticate and get backend
+    backend = get_backend_from_token(request.headers.get("Authorization"))
+
+    # We trust the username supplied by the backend because it is authenticated.
```

Comment:
Drive by comment but I found this line confusing because you can read "it" as referring to the username and then I wondered how that was authenticated.

Maybe a clearer way of putting it would be:
```py
# The request is from an authenticated backend so we trust it to supply arbitrary usernames
```

Reply:
Fixed